sentry-opentelemetry/sentry-opentelemetry-core/src/main/java/io/sentry/opentelemetry/SentrySpanProcessor.java

```diff
@@ -22,14 +98,140 @@ public boolean isStartRequired() {
   }
 
   @Override
-  public void onEnd(ReadableSpan span) {
-    System.out.println(
-        "hello from onEnd" + Thread.currentThread().getId() + Sentry.getCurrentHub().toString());
-    // TODO end
+  public void onEnd(final @NotNull ReadableSpan otelSpan) {
+    if (!hasSentryBeenInitialized()) {
+      return;
+    }
+
+    if (!Instrumenter.OTEL.equals(Sentry.getCurrentHub().getOptions().getInstrumenter())) {
+      return;
+    }
+
+    final @NotNull SpanContext otelSpanContext = otelSpan.getSpanContext();
+    if (!otelSpanContext.isValid()) {
+      return;
+    }
+
+    final @NotNull TraceData traceData = getTraceData(otelSpan);
+    final @Nullable ISpan sentrySpan = spans.remove(traceData.getSpanId());
+
+    if (sentrySpan == null) {
+      return;
+    }
+
+    if (isSentryRequest(otelSpan)) {
```

Comment:
This might be an unnecessary optimization, but I guess we can maybe skip this? As if that is a sentry request, the `sentrySpan` will always be `null` because if was never saved to the map, thus this branch _should_ not really be reachable? (Can be left of course as added security, I guess the overhead is not really relevant, but 🤷 )

Reply:
Hmm agree that it "should" not be reached. If in any case we're unable to detect a request to Sentry in `onStart` I'd rather be safe than sorry though as it would then cause an infinite loop which this `if` is able to avoid. My gut tells me to leave it in.